pkg/datasource.go

```diff
@@ -36,6 +36,7 @@ const LogSearchResponseField_timestamp = "timestamp"
 const MaxLogMetricsDataPoints = 10
 const DefaultLogMetricsDataPoints = 5
 const MinLogMetricsDataPoints = 2
+const numMaxResults = 20000
```

Comment:
It can take a very long time for the plugin to retrieve 10,000 rows (20 pages) from the OCI Logging service due to the fact that each page retrieval via the Logging Search API takes 1+ seconds. So I have several suggestions on how this performance situation might be improved:
1. Increase the number of records retrieved per page, say 1,000,  while reducing the number of pages, say 10.
- I don't know whether this will improve performance but it is worth a shot.
2. Allow the user to control the number of log records returned through the Max Data Points field under Query Options on a data panel, see image below.
- This would allow the user to determine how many log records they want returned based on their use case and how long they are willing to wait for the data to be rendered.
3. Perhaps there is a way to provide the intermediate log record results from the plugin backend to the plugin frontend, such as having the plugin support the streaming capability, but that would take more work.
![QueryOptions-MaxDataPoints](https://user-images.githubusercontent.com/37004987/177006675-d3418579-4da5-410c-bc67-40228d084f49.png)

Reply:
I tested the two situation, with limit 1000 and 10 pages, and limit 500 20 pages, the results are more or less the same:
logger=plugin.oci-logs-datasource t=2022-07-04T12:18:31.51+0000 lvl=dbug msg="Processing log records search query"
logger=plugin.oci-logs-datasource t=2022-07-04T12:18:38.71+0000 lvl=dbug msg="numpage :" numpage=20
logger=plugin.oci-logs-datasource t=2022-07-04T12:18:38.71+0000 lvl=dbug msg="indexCountPag :" indexCountPag=10000


logger=plugin.oci-logs-datasource t=2022-07-04T12:25:17.67+0000 lvl=dbug msg="Processing log records search query
logger=plugin.oci-logs-datasource t=2022-07-04T12:25:24.31+0000 lvl=dbug msg="numpage :" numpage=10
logger=plugin.oci-logs-datasource t=2022-07-04T12:25:24.31+0000 lvl=dbug msg="indexCountPag :" indexCountPag=10000

I think this is indeed a lag in the visualization part, because the data are retrieved by the backend in 7 seconds.